finch/processes/ensemble_utils.py

```diff
@@ -441,6 +405,13 @@ def make_file_groups(files_list: List[Path]) -> List[Dict[str, Path]]:
 
 def make_ensemble(files: List[Path], percentiles: List[int]) -> None:
     ensemble = ensembles.create_ensemble(files)
+    # Depending on the datasets, I've found that writing the netcdf could hang
+    # if the dataset was not loaded explicitely previously... Not sure why.
+    # The datasets should be pretty small when computing the ensembles, so this is
+    # a best effort at working around what looks like a bug in either xclim or xarray.
+    # The xarray documentation mentions: 'this method can be necessary when working
+    # with many file objects on disk.'
+    ensemble.load()
```

Comment:
I guess this is ok if subsets are relatively small but would likely 1) take a long time 2) use large amts of memory  if regions become very big... 
Would a .load() on or after line 417 work as well?
i.e.
```
ensemble_percentiles = ensembles.ensemble_percentiles(ensemble, values=percentiles).load()

```

Reply:
This .load() should use less memory as we are collapsing the 24 models down to 3 variables

Reply:
Could we also maybe try to use 'persist'? 
http://xarray.pydata.org/en/stable/generated/xarray.Dataset.persist.html

Reply:
Good observations, I will look into this.

Reply:
I tried using `persist` and it doesn't work. Looking at the code, it seems to only change datasets containing dask dataframes.

Delaying the call to `.load()` like you mentioned worked though... so it should use less memory. 

For what it's worth, I experienced this process hang only with rcp26 ensembles with the 24 models.

Reply:
Ok sounds good. I'm ok with the changes so feel free to merge